server/src/test/java/org/elasticsearch/search/SearchServiceSingleNodeTests.java

```diff
@@ -263,7 +265,10 @@ public void onQueryPhase(SearchContext context, long tookInNanos) {
 
     @Override
     protected Settings nodeSettings() {
-        return Settings.builder().put("search.default_search_timeout", "5s").build();
+        return Settings.builder()
+            .put("search.default_search_timeout", "5s")
+            .put("thread_pool.search.size", SEARCH_POOL_SIZE) // customized search pool size, reconfiguring at runtime is unsupported
+            .build();
```

Comment:
>  I think that without adjusting the max pool size, we may not be testing what we want to test here, because the creation of slices to parallelize search execution depends on max pool size (as well as other details)

@javanna max (=core) pool size is adjusted by means of above. This is now done deterministically for all tests, previously the change applied to all tests running after `testSlicingBehaviourForParallelCollection`.